include/http.h

```diff
@@ -185,7 +188,14 @@ struct Request {
     return url;
   }
 
+  int ProgressCallback(double dltotal, double dlnow, double ultotal, double ulnow);
+  double GetUploadSpeed() { return upload_speed; }
+  double GetUploadedSize() { return uploaded_size; }
+
  private:
+  double uploaded_size = 0;
+  double upload_speed = 0;
```

Comment:
You would need to accept an optional callback function to be called to provide progress information like `curlpp` has. Speed and other calculations should be done at caller side.

Reply:
You means I should add flag to control the execution of callback funtion

Reply:
The calculation did here just because the upload progress is splited into multi-parts, and must be cumulative addition of every piece size if I want to get the file uploading progress.

Reply:
For multipart uploads, you would need to have a wrapper function to supplied progress function. In the wrapper function, you track each part's upload metrics and send to the supplied progress function.

Reply:
> I am saying this implementation is wrong. You just need to expose an optional callback function for progress like https://github.com/minio/minio-cpp/blob/main/include/args.h#L147

I got it.

Reply:
> For multipart uploads, you would need to have a wrapper function to supplied progress function. In the wrapper function, you track each part's upload metrics and send to the supplied progress function.

ok, I'll modify like this